### PR TITLE
Docs: Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by mgol**
~~NOTE: Wait for https://github.com/jquery/jquery/pull/5564 to land before merging this PR.~~

Migrate the jQuery Core CONTRIBUTING.md, applying Migrate-specific changes.

Fixes gh-486
Fixes gh-492
Ref jquery/jquery#5564

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,87 +1,72 @@
...
 
-The jQuery core development team frequently tracks posts on the [jQuery Development Forum](http://forum.jquery.com/developing-jquery-core). If you have longer posts or questions please feel free to post them there. If you think you've found a bug please [file it in the bug tracker](#How-to-Report-Bugs).
+When opening a pull request, you'll be asked to sign our Contributor License Agreement. Both the Corporate and Individual agreements can be [previewed on GitHub](https://github.com/openjs-foundation/easycla).
 
-Additionally most of the jQuery core development team can be found in the [#jquery-dev](http://webchat.freenode.net/?channels=jquery-dev) IRC channel on irc.freenode.net.
+If you're looking for some good issues to start with, [here are some issues labeled "help wanted" or "patch welcome"](https://github.com/jquery/jquery-migrate/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22%2C%22Patch+Welcome%22).
```
Comment (E-pic-C) [inline CONTRIBUTING.md:19]:
```suggestion
If you're looking for some good issues to start with, [here are some issues labeled "help wanted" or "patch welcome"](https://github.com/jquery/jquery-migrate/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22%2C%22Patch+Welcome%22).
```

Comment (E-pic-C) [reply CONTRIBUTING.md:19]: @e-pic-c

Comment (mgol) [reply CONTRIBUTING.md:19]: @E-pic-C what's the difference?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -93,111 +78,100 @@ Change directory to the newly created dir `jquery-migrate/`:
 $ cd jquery-migrate
 ```
 
-Add the jquery-migrate repository as a remote; I label mine "upstream":
+Add the jQuery Migrate `main` as a remote. I label mine `upstream`:
```
Comment (timmywil) [inline CONTRIBUTING.md:81]:
```suggestion
Add the jQuery Migrate `main` as a remote (e.g. `upstream`):
```

**Review by timmywil — APPROVED** [review]

**Pull request merged**